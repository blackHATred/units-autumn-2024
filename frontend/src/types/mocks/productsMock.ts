import { Product } from '../Product';

export const productsMock: Product[] = [
    {
        id: 1,
        name: 'IPhone 14 Pro',
        description: 'Latest iphone, buy it now',
        price: 999,
        priceSymbol: '$',
        category: 'Электроника',
        imgUrl: '/iphone.png',
    },
    {
        id: 2,
        name: 'Костюм гуся',
        description: 'Запускаем гуся, работяги',
        price: 1000,
        priceSymbol: '₽',
        category: 'Одежда',
    },
    {
        id: 3,
        name: 'Настольная лампа',
        description: 'Говорят, что ее использовали в pixar',
        price: 699,
        category: 'Для дома',
        imgUrl: '/lamp.png',
    },
    {
        id: 4,
        name: 'Принтер',
        description: 'Незаменимая вещь для студента',
        price: 7000,
        category: 'Электроника',
    },
];
