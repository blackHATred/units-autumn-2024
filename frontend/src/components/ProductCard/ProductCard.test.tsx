import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ProductCard } from './ProductCard';
import { productsMock } from '../../types/mocks/productsMock';

describe('ProductCard test', () => {
    let rendered: RenderResult;

    it('should render correctly', () => {
        rendered = render(<ProductCard {...productsMock[0]} />);
        expect(rendered.asFragment()).toMatchSnapshot();
    });

    it('should render correctly without image', () => {
        rendered = render(<ProductCard {...productsMock[1]} />);
        expect(rendered.asFragment()).toMatchSnapshot();
    });

    it('should render correctly without price symbol', () => {
        rendered = render(<ProductCard {...productsMock[2]} />);
        expect(rendered.asFragment()).toMatchSnapshot();
    });
});
