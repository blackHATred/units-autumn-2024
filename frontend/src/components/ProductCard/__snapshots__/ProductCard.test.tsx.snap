// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard test should render correctly 1`] = `
<DocumentFragment>
  <div
    class="product-card"
    data-testid="product_container"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        IPhone 14 Pro
      </h2>
      <div
        class="product-card__description"
      >
        Latest iphone, buy it now
      </div>
      <h1
        class="product-card__price"
      >
        999 $
      </h1>
      <div
        class="product-card__category"
      >
        Электроника
      </div>
    </div>
    <img
      alt="IPhone 14 Pro"
      class="product-card__image"
      src="/iphone.png"
    />
  </div>
</DocumentFragment>
`;

exports[`ProductCard test should render correctly without image 1`] = `
<DocumentFragment>
  <div
    class="product-card"
    data-testid="product_container"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        Костюм гуся
      </h2>
      <div
        class="product-card__description"
      >
        Запускаем гуся, работяги
      </div>
      <h1
        class="product-card__price"
      >
        1 000 ₽
      </h1>
      <div
        class="product-card__category"
      >
        Одежда
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProductCard test should render correctly without price symbol 1`] = `
<DocumentFragment>
  <div
    class="product-card"
    data-testid="product_container"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        Настольная лампа
      </h2>
      <div
        class="product-card__description"
      >
        Говорят, что ее использовали в pixar
      </div>
      <h1
        class="product-card__price"
      >
        699 ₽
      </h1>
      <div
        class="product-card__category"
      >
        Для дома
      </div>
    </div>
    <img
      alt="Настольная лампа"
      class="product-card__image"
      src="/lamp.png"
    />
  </div>
</DocumentFragment>
`;
